docs/source/notes/libtorch_stable_abi.md

```diff
@@ -0,0 +1,48 @@
+# LibTorch Stable ABI
+
+This note will eventually contain more details on how to use the APIs in torch/csrc/stable. For the moment, it contains a table of internal representations:
+1. type in custom extension: type used within the end user custom library.
+2. StableIValue representation: a stable conversion of the type to liaison between the user model vs libtorch.so.
+3. type in libtorch: type used within libtorch.so.
+4. Schema Type: type as described by the schema, which we hail as the source of truth for both ATen ops in native_functions.yaml and for user defined custom ops registered to the dispatcher.
+
+|  type in custom extension    |   StableIValue representation   |   type in libtorch  |   Schema Type  |
+| -------- | ------- | ------- | ------- |
+| std::optional\<S> | \*reinterpret_cast\<(StableIValue\*)\*>, pointer to a StableIValue recursively defined | std::optional\<T> | OptionalType |
+| std::nullopt | \*reinterpret_cast\<nullptr_t\*> | IValue() | NoneType |
+| RAIIATH | \*reinterpret_cast\<uint64_t\*> of AtenTensorHandle | at::Tensor |  TensorType |
```

Comment:
<img width="861" alt="image" src="https://github.com/user-attachments/assets/6a092e91-4fc1-4cf7-a7ae-95880fc18ec0" />
Are all the *'s intentional? I'm having a hard time parsing this

Reply:
The *s are intentional--we're type punning and casting to avoid compiler warnings